### PR TITLE
Changed signup routing

### PR DESCRIPTION
**Pull request by Daniel-J-Burgess**
Split the signup pages, fixed a few bugs and added pin to receptionist signup.

### CODE REVIEW

File: client/__tests__/__mocks__/signup.mock.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export const validSignup = [{
     request: {
         query: gql`
             mutation {
-                signup(email: "test@mail.com", password: "password1!", type: "resident", idNumber: "1")
+                signup(email: "test@mail.com", password: "password1!",confirmationPin:"11111", type: "resident", idNumber: "1")
```
Comment (LarisaBothaTuks) [inline client/__tests__/__mocks__/signup.mock.js:8]: why is it hardcoded?

File: backend/libs/auth/src/auth.service.spec.ts [hunk 2]
```diff
@@ -220,7 +220,7 @@ describe("AuthService", () => {
             const findOneMock = jest.spyOn(userService, 'findOne').mockReturnValueOnce(Promise.resolve(null))
             // Act
             try {
-                const response = await service.signup({ email: 'mail', password: 'password', type: 'receptionist', idNumber: 2323 })
+                const response = await service.signup({ email: 'mail', password: 'password',confirmationPin:'00000', type: 'receptionist', idNumber: 2323 })
```
Comment (LarisaBothaTuks) [inline backend/libs/auth/src/auth.service.spec.ts:223]: there is alot of unused variables on this page

File: client/components/receptionistSignButton.jsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ const ReceptionistSignButton = ({ htmlFor, text, colour, onClick, key, signInTim
         >
             {text}
             {signInTime != null ? (
-                <div class="inline-flex absolute -top-2 -right-2 justify-center border-none bg-secondary items-center w-6 h-6 text-xs font-bold text-white rounded-full">{(new Date()).getDate()-(new Date(signInTime.replace(/-/gi,"/")).getDate())}</div>
+                <div className="inline-flex absolute -top-2 -right-2 justify-center border-none bg-secondary items-center w-6 h-6 text-xs font-bold text-white rounded-full">{(new Date()).getDate()-(new Date(signInTime.replace(/-/gi,"/")).getDate())}</div>
```
Comment (LarisaBothaTuks) [inline client/components/receptionistSignButton.jsx:11]: thank you

File: client/pages/_app.js [hunk 4]
```diff
@@ -41,7 +41,9 @@ function MyApp({ Component, pageProps }) {
             "/login",
             "/",
             "/expire",
-            "/signUp",
+            "/residentSignUp",
+            "/receptionistSignUp",
+            "/signUpSelect",
```
Comment (LarisaBothaTuks) [inline client/pages/_app.js:46]: change signUp select to signup please?

File: client/store/authStore.js [hunk 5]
```diff
@@ -76,7 +76,8 @@ const useAuth = create(
                 } else {
                     return [
                         { content: "Login", path: "/login" },
-                        { content: "Signup", path: "/signUp"}
+                        { content: "SignUpSelect", path: "/signUpSelect"}
```
Comment (LarisaBothaTuks) [inline client/store/authStore.js:79]: content: SignUp should fix the issue if im understanding correctly

**Review by LarisaBothaTuks — COMMENTED** [review]
Minor fixes, please

File: client/__tests__/__mocks__/signup.mock.js (hunk 1 above)
Comment (Daniel-J-Burgess) [reply client/__tests__/__mocks__/signup.mock.js:8]: We currently are not giving the residents a pin however should we choose to do so is setup in a way to make it possible.

**Review by stefan-van-schoor — APPROVED** [review]

**Pull request merged**